pinot-core/src/main/java/org/apache/pinot/core/query/postaggregation/PostAggregationFunction.java

```diff
@@ -36,6 +36,8 @@
   private final ColumnDataType _resultType;
 
   public PostAggregationFunction(String functionName, ColumnDataType[] argumentTypes) {
+    Preconditions.checkArgument(FunctionRegistry.containsFunction(functionName), "Unsupported function: %s not found", 
```

Comment:
Suggest expanding the existing preconditions check similar to how it is handled in `TransformFunctionFactory` to avoid the unnecessary overhead
```suggestion
    if (functionInfo == null) {
      if (FunctionRegistry.containsFunction(functionName)) {
        throw new IllegalArgumentException(...);
      } else {
        throw new IllegalArgumentException(...);
      }
    }
```

Reply:
Addressed in https://github.com/apache/pinot/commit/f02a015291428e62f7f3dc31b10a323844a4e447